examples/shader/post_processing.rs

```diff
@@ -162,20 +78,254 @@ fn main_camera_cube_rotator_system(
+                    let size = {
+                        let window = windows.get(fit_to_window.window_id).expect("PostProcessingCamera is rendering to a window, but this window could not be found");
+                        Extent3d {
+                            width: window.physical_width(),
+                            height: window.physical_height(),
+                            ..Default::default()
+                        }
+                    };
+                    let image = images.get_mut(&fit_to_window.image).expect(
+                        "FitToWindowSize is referring to an Image, but this Image could not be found",
+                    );
+                    info!("resize to {:?}", size);
+                    image.resize(size);
+                    // Hack because of https://github.com/bevyengine/bevy/issues/5595
+                    image_events.send(AssetEvent::Modified {
+                        handle: fit_to_window.image.clone(),
+                    });
+                    post_processing_materials.get_mut(&fit_to_window.material);
+                }
+            }
+        }
+    }
+
+    fn update_material(
+        time: Res<Time>,
+        cameras: Query<&Handle<PostProcessingMaterial>>,
+        mut materials: ResMut<Assets<PostProcessingMaterial>>,
+    ) {
+        for handle in &cameras {
+            let mut mat = materials.get_mut(handle).unwrap();
+
+            mat.offset_r = Vec2::new(-0.01f32 * time.seconds_since_startup().sin() as f32, 0f32);
+            mat.offset_g = Vec2::new(
+                0.02f32 * time.seconds_since_startup().sin() as f32,
+                0.02f32 * time.seconds_since_startup().cos() as f32,
+            );
+            mat.offset_b = Vec2::new(0f32, -0.01f32 * time.seconds_since_startup().cos() as f32);
+        }
+    }
+
+    /// sets up post processing for cameras that have had `PostProcessingCamera` added
+    fn setup_new_post_processing_cameras(
+        mut commands: Commands,
+        windows: Res<Windows>,
+        mut meshes: ResMut<Assets<Mesh>>,
+        mut post_processing_materials: ResMut<Assets<PostProcessingMaterial>>,
+        mut images: ResMut<Assets<Image>>,
+        mut cameras: Query<(Entity, &mut Camera), Added<PostProcessingCamera>>,
+    ) {
+        for (entity, mut camera) in &mut cameras {
+            let original_target = camera.target.clone();
+
+            let mut option_window_id: Option<WindowId> = None;
+
+            // Get the size the camera is rendering to
+            let size = match &camera.target {
+                RenderTarget::Window(window_id) => {
+                    let window = windows.get(*window_id).expect("PostProcessingCamera is rendering to a window, but this window could not be found");
+                    option_window_id = Some(*window_id);
+                    Extent3d {
+                        width: window.physical_width(),
+                        height: window.physical_height(),
```

Comment:
This is incorrect and doesn't work with UI scaling. The correct solution is either

```rust
let scale = window.scale_factor();

// ...

width: (window.physical_width() as f64 / scale) as u32,
height: (window.physical_height() as f64 / scale) as u32,
```

or just the simpler alternative that already returns the logical width

```rust
width: window.width() as u32,
height: window.height() as u32,
```